From c9bb379c9959ae9db3799ebf7a8283f5b1e79753 Mon Sep 17 00:00:00 2001
From: shixuantong <shixuantong1@huawei.com>
Date: Fri, 28 Nov 2025 16:53:24 +0800
Subject: [PATCH] fix syntar in sentryctl

---
 src/services/syssentry/sentryctl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/services/syssentry/sentryctl b/src/services/syssentry/sentryctl
index 1e00252..43832f8 100644
--- a/src/services/syssentry/sentryctl
+++ b/src/services/syssentry/sentryctl
@@ -244,7 +244,7 @@ if __name__ == '__main__':
             {"name": "server_cna", "type": str, "choices": None, "required" : True, "help": "server cna array"},
         ],
         "sentry_reporter" : [
-            {"name" : "ub_mem_fault_with_kill", "type" : "str", "choices":  ["on", "off"], "required" : False, "help" : "Enable/Disable sending SIGBUS signal with UB mem event"},
+            {"name" : "ub_mem_fault_with_kill", "type" : str, "choices":  ["on", "off"], "required" : False, "help" : "Enable/Disable sending SIGBUS signal with UB mem event"},
             {"name" : "ub_mem_fault", "type" : str, "choices": ["on", "off"], "required" : False, "help": "Enable/Disable UB mem event"},
             {"name" : "power_off", "type": str, "choices": ["on", "off"], "required" : False, "help": "Enable/Disable power off event"},
             {"name" : "oom", "type" : str, "choices": ["on", "off"], "required" : False, "help": "Enable/Disable oom event"},
-- 
2.27.0

