From 37c606446a45fedd08ff3782e0185552f8f34dec Mon Sep 17 00:00:00 2001
From: shixuantong <shixuantong1@huawei.com>
Date: Tue, 9 Dec 2025 11:15:22 +0800
Subject: [PATCH] fix potential use-after-free bugs in libxalarm

---
 src/libs/libxalarm/register_xalarm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libs/libxalarm/register_xalarm.c b/src/libs/libxalarm/register_xalarm.c
index 13584cb..0309ae6 100644
--- a/src/libs/libxalarm/register_xalarm.c
+++ b/src/libs/libxalarm/register_xalarm.c
@@ -676,6 +676,7 @@ int xalarm_register_event(struct alarm_register **register_info, struct alarm_su
     (*register_info)->register_fd = create_unix_socket(PATH_REG_ALARM);
     if ((*register_info)->register_fd == -1) {
         free(*register_info);
+        *register_info = NULL;
         return -ENOTCONN;
     }
 
-- 
2.27.0

