From eca8c542875aef5cfbf947d697c4b644490d1c05 Mon Sep 17 00:00:00 2001
From: zhuofeng <zhuofeng2@huawei.com>
Date: Fri, 30 Aug 2024 19:58:41 +0800
Subject: [PATCH] add deleted code to plugin rasdaemon

---
 src/python/syssentry/syssentry.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/python/syssentry/syssentry.py b/src/python/syssentry/syssentry.py
index 32b81e3..3d5cb8d 100644
--- a/src/python/syssentry/syssentry.py
+++ b/src/python/syssentry/syssentry.py
@@ -462,6 +462,14 @@ def main_loop():
     epoll_fd.register(cpu_alarm_fd.fileno(), select.EPOLLIN)
 
     logging.debug("start main loop")
+    # onstart_tasks_handle()
+    for task_type in TasksMap.tasks_dict:
+        for task_name in TasksMap.tasks_dict.get(task_type):
+            task = TasksMap.tasks_dict.get(task_type).get(task_name)
+            if not task:
+                continue
+            task.onstart_handle()
+
     while True:
         try:
             events_list = epoll_fd.poll(SERVER_EPOLL_TIMEOUT)
-- 
2.33.0

