From 797f0760d3dc1e396547edffad887e273c5731c6 Mon Sep 17 00:00:00 2001
From: shixuantong <shixuantong1@huawei.com>
Date: Tue, 25 Nov 2025 16:42:28 +0800
Subject: [PATCH 14/16] fix Out of memory bounds access in ebpf_collector

---
 .../sentryCollector/ebpf_collector/ebpf_collector.c        | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/services/sentryCollector/ebpf_collector/ebpf_collector.c b/src/services/sentryCollector/ebpf_collector/ebpf_collector.c
index 445cce7..e000f0f 100644
--- a/src/services/sentryCollector/ebpf_collector/ebpf_collector.c
+++ b/src/services/sentryCollector/ebpf_collector/ebpf_collector.c
@@ -240,7 +240,12 @@ char *read_config_value(const char *file, const char *section, const char *key)
     char *value = NULL;
 
     while (fgets(line, sizeof(line), fp) != NULL) {
-        line[strcspn(line, "\n")] = 0;
+        size_t newline_char_idx = strcspn(line, "\n");
+        if (newline_char_idx == MAX_LINE_LENGTH) {
+            line[MAX_LINE_LENGTH - 1] = '\0';
+        } else {
+            line[newline_char_idx] = '\0';
+        }
 
         if (line[0] == '\0' || line[0] == ';' || line[0] == '#') {
             continue;
-- 
2.27.0

