From d4f28fadee2d536699e21fca49e8e14782f4d19c Mon Sep 17 00:00:00 2001
From: caixiaomeng 00662745 <caixiaomeng2@huawei.com>
Date: Tue, 18 Mar 2025 20:44:46 +0800
Subject: [PATCH 2/7] fix xalarmd stop failed by systemd

---
 config/service/xalarmd.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/service/xalarmd.service b/config/service/xalarmd.service
index 20db997..6d8418c 100644
--- a/config/service/xalarmd.service
+++ b/config/service/xalarmd.service
@@ -4,7 +4,7 @@ Description = xalarm daemon for alarm messages forwarding
 [Service]
 Type = forking
 ExecStart=/usr/bin/python3 /usr/bin/xalarmd
-ExecStop=/bin/kill 
+ExecStop=/bin/kill $MAINPID
 KillMode=process
 Restart=on-failure
 RestartSec=3s
-- 
2.43.0

