From 435f85059f5c4adb5d237b5ba8e16467ed4889b0 Mon Sep 17 00:00:00 2001
From: shixuantong <shixuantong1@huawei.com>
Date: Tue, 17 Mar 2026 09:59:14 +0800
Subject: [PATCH 08/17] fix memset usage

---
 src/libs/libxalarm/register_xalarm.c | 21 ++++-----------------
 1 file changed, 4 insertions(+), 17 deletions(-)

diff --git a/src/libs/libxalarm/register_xalarm.c b/src/libs/libxalarm/register_xalarm.c
index a451000..5fbb5a3 100644
--- a/src/libs/libxalarm/register_xalarm.c
+++ b/src/libs/libxalarm/register_xalarm.c
@@ -106,10 +106,7 @@ static int create_unix_socket(const char *path)
         goto release_socket;
     }
 
-    if (memset(&alarm_addr, 0, sizeof(alarm_addr)) == NULL) {
-        printf("create_unix_socket:  memset alarm_addr failed, ret: %d\n", ret);
-        goto release_socket;
-    }
+    memset(&alarm_addr, 0, sizeof(alarm_addr));
     alarm_addr.sun_family = AF_UNIX;
     strncpy(alarm_addr.sun_path, path, sizeof(alarm_addr.sun_path) - 1);
 
@@ -313,10 +310,7 @@ static int init_report_addr(struct sockaddr_un *alarm_addr, char *report_path)
         return -1;
     }
 
-    if (memset(alarm_addr, 0, sizeof(struct sockaddr_un)) == NULL) {
-        fprintf(stderr, "%s: memset  alarm_addr failed\n", __func__);
-        return -1;
-    }
+    memset(alarm_addr, 0, sizeof(struct sockaddr_un));
     alarm_addr->sun_family = AF_UNIX;
     strncpy(alarm_addr->sun_path, report_path, sizeof(alarm_addr->sun_path) - 1);
 
@@ -342,10 +336,7 @@ int xalarm_Report(unsigned short usAlarmId, unsigned char ucAlarmLevel,
         return -1;
     }
 
-    if (memset(&info, 0, sizeof(struct alarm_info)) == NULL) {
-        fprintf(stderr, "%s: memset info failed\n", __func__);
-        return -1;
-    }
+    memset(&info, 0, sizeof(struct alarm_info));
     info.usAlarmId = usAlarmId;
     info.ucAlarmLevel = ucAlarmLevel;
     info.ucAlarmType = ucAlarmType;
@@ -518,11 +509,7 @@ int send_data_to_socket(const char *socket_path, const char *message)
     }
 
     // set socket address
-    if (memset(&addr, 0, sizeof(struct sockaddr_un)) == NULL) {
-        fprintf(stderr, "%s: memset info failed.\n", __func__);
-        close(sockfd);
-        return RETURN_CODE_FAIL;
-    }
+    memset(&addr, 0, sizeof(struct sockaddr_un));
 
     addr.sun_family = AF_UNIX;
     strncpy(addr.sun_path, socket_path, sizeof(addr.sun_path) - 1);
-- 
2.27.0

