From 2f0a7ae0939ba21df2fcb5d4648fa2136402dd6e Mon Sep 17 00:00:00 2001
From: shixuantong <shixuantong1@huawei.com>
Date: Tue, 17 Mar 2026 19:49:35 +0800
Subject: [PATCH 17/17] fix NameError in task_get_alarm()

---
 src/services/syssentry/callbacks.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/services/syssentry/callbacks.py b/src/services/syssentry/callbacks.py
index ac9f0ee..efcd887 100644
--- a/src/services/syssentry/callbacks.py
+++ b/src/services/syssentry/callbacks.py
@@ -54,12 +54,15 @@ def task_get_result(mod_name):
 def task_get_alarm(data):
     """get alarm by mod name"""
     try:
+        if not isinstance(data, dict):
+            return "failed", "data type error, current type is %s" % type(dict)
+        task_name, time_range, detailed = None, None, None
         task_name = data['task_name']
         time_range = data['time_range']
         detailed = data['detailed']
     except KeyError:
-        logging.debug("Key 'detailed' does not exist in the dictionary")
-        detailed = None
+        logging.debug("the expected key does not exist in the dictionary")
+
     task = TasksMap.get_task_by_name(task_name)
     if not task:
         return "failed", f"cannot find task by name {task_name}"
-- 
2.27.0

