From 7baf2815f515c54bc33f41f495ec7c26988b5c44 Mon Sep 17 00:00:00 2001
From: shixuantong <shixuantong1@huawei.com>
Date: Tue, 11 Jun 2024 16:47:46 +0800
Subject: [PATCH] fix version in setup.py

---
 src/python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/python/setup.py b/src/python/setup.py
index 21dbe9f..f96a96e 100644
--- a/src/python/setup.py
+++ b/src/python/setup.py
@@ -17,7 +17,7 @@ from setuptools import setup, find_packages
 
 setup(
     name="syssentry",
-    version="1.0.1",
+    version="1.0.2",
     description="System inspection framework tool set",
     packages=find_packages(),
     include_package_data=True,
-- 
2.27.0

