From 85d23aa0f18bd79a54d247f39503a49a9e4fa772 Mon Sep 17 00:00:00 2001
From: hubin <hubin73@huawei.com>
Date: Sat, 22 Mar 2025 16:41:55 +0800
Subject: [PATCH 4/7] fix missing pycache file

---
 Makefile | 21 +++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 330a7d4..01ed310 100644
--- a/Makefile
+++ b/Makefile
@@ -68,15 +68,17 @@ dirs:
 
 isentry:
 	cd $(CURSRCDIR) && $(PYBIN) setup.py install --prefix=$(PREFIX) -O1 --root=build --record=SENTRY_FILES
-	find $(CURSRCDIR)/build/ -type d -name '__pycache__' -exec rm -rf {} +
 	
 	# sysSentry主包
 	## 创建目录
 	install -d -m 700 $(LOGSAVEDIR)/sysSentry
 	install -d -m 700 $(VARLIB)/logrotate-syssentry
 	install -d -m 700 $(PYDIR)/syssentry
+	install -d -m 700 $(PYDIR)/syssentry/__pycache__
 	install -d -m 755 $(PYDIR)/xalarm
+	install -d -m 700 $(PYDIR)/xalarm/__pycache__
 	install -d -m 700 $(PYDIR)/sentryCollector
+	install -d -m 700 $(PYDIR)/sentryCollector/__pycache__
 	install -d -m 700 $(PYDIR)/$(PKGVEREGG)
 	install -d -m 700 $(ETCDIR)/sysconfig
 	
@@ -95,9 +97,12 @@ isentry:
 	install -m 600 $(CURCONFIGDIR)/service/sentryCollector.service $(SYSTEMDDIR)
 	
 	## 安装python源代码文件到相应的目录
-	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/syssentry/* $(PYDIR)/syssentry
-	install -m 555 src/build/usr/lib/$(PYNAME)/site-packages/services/xalarm/* $(PYDIR)/xalarm
-	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/sentryCollector/* $(PYDIR)/sentryCollector
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/syssentry/*.py $(PYDIR)/syssentry
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/syssentry/__pycache__/* $(PYDIR)/syssentry/__pycache__
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/xalarm/*.py $(PYDIR)/xalarm
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/xalarm/__pycache__/* $(PYDIR)/xalarm/__pycache__
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/sentryCollector/*.py $(PYDIR)/sentryCollector
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/services/sentryCollector/__pycache__/* $(PYDIR)/sentryCollector/__pycache__
 	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/$(PKGVEREGG)/* $(PYDIR)/$(PKGVEREGG)
 	
 	## 安装可执行文件
@@ -111,15 +116,19 @@ isentry:
 	
 	# avg_block_io
 	install -d -m 700 $(PYDIR)/sentryPlugins/avg_block_io
+	install -d -m 700 $(PYDIR)/sentryPlugins/avg_block_io/__pycache__
 	install -m 550 $(CURSRCDIR)/build/usr/bin/avg_block_io $(BINDIR)
-	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/sentryPlugins/avg_block_io/* $(PYDIR)/sentryPlugins/avg_block_io
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/sentryPlugins/avg_block_io/*.py $(PYDIR)/sentryPlugins/avg_block_io
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/sentryPlugins/avg_block_io/__pycache__/* $(PYDIR)/sentryPlugins/avg_block_io/__pycache__
 	install -m 600 $(CURCONFIGDIR)/plugins/avg_block_io.ini $(ETCDIR)/sysSentry/plugins/
 	install -m 600 $(CURCONFIGDIR)/tasks/avg_block_io.mod $(ETCDIR)/sysSentry/tasks/
 
 	# ai_block_io
 	install -d -m 700 $(PYDIR)/sentryPlugins/ai_block_io
+	install -d -m 700 $(PYDIR)/sentryPlugins/ai_block_io/__pycache__
 	install -m 550 $(CURSRCDIR)/build/usr/bin/ai_block_io $(BINDIR)
-	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/sentryPlugins/ai_block_io/* $(PYDIR)/sentryPlugins/ai_block_io
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/sentryPlugins/ai_block_io/*.py $(PYDIR)/sentryPlugins/ai_block_io
+	install -m 550 src/build/usr/lib/$(PYNAME)/site-packages/sentryPlugins/ai_block_io/__pycache__/* $(PYDIR)/sentryPlugins/ai_block_io/__pycache__
 	install -m 600 $(CURCONFIGDIR)/plugins/ai_block_io.ini $(ETCDIR)/sysSentry/plugins/
 	install -m 600 $(CURCONFIGDIR)/tasks/ai_block_io.mod $(ETCDIR)/sysSentry/tasks/
 
-- 
2.43.0

