From 3fdf5ec519893387fd89a8cb3faa34213f935cd2 Mon Sep 17 00:00:00 2001
From: shixuantong <shixuantong1@huawei.com>
Date: Thu, 6 Nov 2025 10:25:45 +0800
Subject: [PATCH 06/16] Fix the use of uninitialized variable 'ret'

---
 src/libs/libxalarm/register_xalarm.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/libs/libxalarm/register_xalarm.c b/src/libs/libxalarm/register_xalarm.c
index f43297a..13584cb 100644
--- a/src/libs/libxalarm/register_xalarm.c
+++ b/src/libs/libxalarm/register_xalarm.c
@@ -312,15 +312,13 @@ char *xalarm_getdesc(const struct alarm_info *palarm)
 
 static int init_report_addr(struct sockaddr_un *alarm_addr, char *report_path)
 {
-    int ret;
-
     if (alarm_addr == NULL) {
         fprintf(stderr, "%s: alarm_addr is null\n", __func__);
         return -1;
     }
 
     if (memset(alarm_addr, 0, sizeof(struct sockaddr_un)) == NULL) {
-        fprintf(stderr, "%s: memset  alarm_addr failed, ret: %d\n", __func__, ret);
+        fprintf(stderr, "%s: memset  alarm_addr failed\n", __func__);
         return -1;
     }
     alarm_addr->sun_family = AF_UNIX;
@@ -332,7 +330,7 @@ static int init_report_addr(struct sockaddr_un *alarm_addr, char *report_path)
 int xalarm_Report(unsigned short usAlarmId, unsigned char ucAlarmLevel,
     unsigned char ucAlarmType, char *pucParas)
 {
-    int ret, fd;
+    int ret = 0, fd;
     struct alarm_info info;
     struct sockaddr_un alarm_addr;
 
@@ -349,7 +347,7 @@ int xalarm_Report(unsigned short usAlarmId, unsigned char ucAlarmLevel,
     }
 
     if (memset(&info, 0, sizeof(struct alarm_info)) == NULL) {
-        fprintf(stderr, "%s: memset info failed, ret: %d\n", __func__, ret);
+        fprintf(stderr, "%s: memset info failed\n", __func__);
         return -1;
     }
     info.usAlarmId = usAlarmId;
-- 
2.27.0

